cluster_configs/barnacle/barnacle_status.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python
+
+import subprocess
+import sys
+import time
+import re
+from io import StringIO
+
+def parse_qstat(lines):
+    q_dict = {}
+
+    for line in lines:
+        if line.startswith(' '):
+            entry = line.rstrip()
+            p = re.compile('^ +(.+?) = (.+)')
+            m = p.match(entry.rstrip())
+            q_dict[m.groups()[0]] = m.groups()[1]
+        elif line.startswith('\t'):
+            entry += line.strip()
+            p = re.compile('^ +(.+?) = (.+)')
+            m = p.match(entry.rstrip())
+            q_dict[m.groups()[0]] = m.groups()[1]
+
+    return(q_dict)
+
+def get_status(q_dict):
+    try:
+        if q_dict['job_state'] == 'R':
+            status = "running"
+        elif q_dict['job_state'] == 'Q':
+            status = "running"
+        elif q_dict['job_state'] == 'C' and q_dict['exit_status'] == '0':
+            status = "success"
+        elif q_dict['job_state'] == 'C' and q_dict['exit_status'] != '0':
+            status = "failed"
+        else:
+            status = "failed"
+    except KeyError:
+        status = "failed"
+
+    return(status)
+
+def main():
+    jobid = sys.argv[1]
+
+    TRY_TIMES = 3
+
+    for i in range(TRY_TIMES):
+        cmd = "qstat -f {}".format(jobid)
```

Comment:
where is format() defined? Is that an alternative way of saying "qstat -f %s" % jobid ?

Reply:
yep, it's filling in the `{}` with `jobid`